colorutils.cpp

```diff
@@ -0,0 +1,40 @@
+#include "colorutils.h"
+#include <iostream>
+#include <cassert>
+#include <iomanip>
+
+namespace TelCoColorCoder {
+    ColorPair GetColorFromPairNumber(int pairNumber) {
+        int zeroBasedPairNumber = pairNumber - 1;
+        MajorColor majorColor = (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+        MinorColor minorColor = (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+        return ColorPair(majorColor, minorColor);
+    }
+
+    int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
+        return major * numberOfMinorColors + minor + 1;
+    }
+
+    void testNumberToPair(int pairNumber, MajorColor expectedMajor, MinorColor expectedMinor) {

```

Comment:
This class mixes test code with production code. People who require functionality (conversion between number and color) will be forced to take the test code also.